framework/src/modules/chain/synchronizer/fast_chain_switching_mechanism.js

```diff
@@ -1,14 +1,59 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
 class FastChainSwitchingMechanism {
-	constructor() {
+	constructor({ storage, logger, slots, dpos, activeDelegates }) {
+		this.storage = storage;
+		this.logger = logger;
+		this.slots = slots;
+		this.dpos = dpos;
+		this.constants = {
+			activeDelegates,
+		};
 		this.active = false;
 	}
 
 	// eslint-disable-next-line class-methods-use-this,no-empty-function
 	async run() {}
 
-	isActive() {
+	get isActive() {
 		return this.active;
 	}
+
+	/**
+	 * Check if this sync mechanism is valid for the received block
+	 *
+	 * @param {object} receivedBlock - The blocked received from the network
+	 * @return {Promise.<Boolean|undefined>} - If the mechanism applied to received block
+	 * @throws {Error} - In case want to abort the sync pipeline
+	 */
+	async isValidFor(receivedBlock) {
+		const lastBlock = await this.storage.entities.Block.getLastBlock();
```

Comment:
it shouldn't access the database to ask for the last block. It should ask the blocks module

Reply:
Why we add dependency of `blocks` module here just to get the last block, which we can easily get from the storage and storage is already here.

Reply:
we should only get the last block from the blocks module because it's managed in memory.
when to update last block is the responsibility of `blocks`
This problem happened already multiple times, so let's consistently use the blocks

Reply:
Our logic here does not refer to something in memory. We need to get last persisted block, for which storage is the right source of truth. As said, I don't want to add a new dependency to blocks just to fetch last persisted block, which is already can be done through storage.